fsharp-backend/tests/testfiles/internal.tests

```diff
@@ -88,5 +112,70 @@ DarkInternal.canvasIdOfCanvasName_v0 "not-a-canvas" = Nothing
  [DarkInternal.checkPermission_v0 user1 user2; startingPermission] ) = ["rw" ; ""]
 
 
-[test.canvasId length]
-(DarkInternal.canvasIdOfCanvasName_v0_ster (Test.getCanvasName)) = toString Test.getCanvasID
+// ---------------
+// canvasID
+// ---------------
+[tests.canvasIDOfCanvasName]
+DarkInternal.canvasIdOfCanvasName_v0 "not-a-canvas" = Nothing
+DarkInternal.canvasIdOfCanvasName_v0_ster Test.getCanvasName = toString Test.getCanvasID
+DarkInternal.canvasIDOfCanvasName_v0 Test.getCanvasName = Ok (Test.getCanvasID)
+DarkInternal.canvasIDOfCanvasName_v0 "invalid name" = Error "Invalid username 'invalid name' - must be 2-20 lowercase characters, and must start with a letter."
+DarkInternal.canvasIDOfCanvasName_v0 "not-a-real-canvas" = Error "Canvas not found"
+
+// ---------------
+// Secrets
+// ---------------
+[tests.secrets]
+
+[test.getSecrets empty]
+DarkInternal.getSecrets Test.getCanvasID = {}
+
+[test.insertSecret]
+(let _ = DarkInternal.insertSecret Test.getCanvasID "SECRET" "VALUE"
+ DarkInternal.getSecrets Test.getCanvasID) = { SECRET = "VALUE" }
```

Comment:
another test to add: try to insert the same secret key twice

Reply:
We probably have that elsewhere in the secrets-specific tests, but would be additionally good to verify how the API handler responds.